src/gg.regression.unity.bots/Runtime/Scripts/ActionManager/Actions/UIInputFieldSubmitAction.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Newtonsoft.Json.Linq;
+using RegressionGames.StateRecorder.JsonConverters;
+using TMPro;
+using UnityEngine;
+using UnityEngine.InputSystem;
+using UnityEngine.UI;
+using Object = UnityEngine.Object;
+
+namespace RegressionGames.ActionManager.Actions
+{
+    /// <summary>
+    /// Action for pressing/releasing the submit key on the focused input field.
+    /// </summary>
+    public class UIInputFieldSubmitAction : RGGameAction
+    {
+        public string NormalizedGameObjectName { get; set; }
+        
+        public UIInputFieldSubmitAction(string[] path, Type objectType, string normalizedGameObjectName) : 
+            base(path, objectType, new RGBoolRange())
+        {
+            Debug.Assert(typeof(InputField).IsAssignableFrom(objectType) || typeof(TMP_InputField).IsAssignableFrom(objectType));
```

Comment:
I haven't seen us use this before. Can you give a short summary of how it works.

Normally for things like this we've written code to manually check and used `RGDebug.LogError` explicitly or thrown an approriate exception depending on the code location

Reply:
Every action that is identified during the static analysis is associated with a particular object/component type. In this case, the assertion is just to serve as a restriction on the types of objects this action can be used with. The actual construction of these actions is all handled within the `RGActionAnalysis` class (i.e. it is not a user-facing API) so I just impose this restriction with a simple assertion to serve as a sanity check that the analysis is not doing something unexpected.

Reply:
I'm unfamiliar with the intricacies of this within C#, but in other languages I like to stay away from Assert-based functionality because it can be turned off. Why not do an explicit check and throw an error? That way in release builds, this would still be caught (in releases, [Debug.Assert is removed](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.debug.assert?view=net-8.0))

Reply:
Normally that's what I would do if these were user-facing APIs, but these checks are just light sanity checks to catch the errors from RGActionAnalysis sooner during development. Anyways, I have no problem with converting these to exceptions - I'll make the change.